datafusion/sqllogictest/test_files/joins.slt

```diff
@@ -4225,7 +4225,7 @@ query IIIIB
 SELECT * FROM t0 FULL JOIN t1 ON t0.c2 >= t1.c2 LIMIT 2;
 ----
 2 2 2 2 true
-3 3 2 2 true
+2 2 2 2 false
```

Comment:
Why this change?

According to this PR, the limit push down to full join both side:
https://github.com/apache/datafusion/pull/12963